src/odfesqlodbc/es_types.c

```diff
@@ -480,35 +479,35 @@ const char *estype_attr_to_name(const ConnectionClass *conn, OID type,
     UNUSED(conn, typmod, conn, auto_increment);
     switch (type) {
         case ES_TYPE_BOOL:
-            return "boolean";
+            return ES_TYPE_NAME_BOOLEAN;
         case ES_TYPE_INT1:
-            return "byte";
+            return ES_TYPE_NAME_BYTE;
         case ES_TYPE_INT2:
-            return "short";
+            return ES_TYPE_NAME_SHORT;
         case ES_TYPE_INT4:
-            return "integer";
+            return ES_TYPE_NAME_INTEGER;
         case ES_TYPE_INT8:
-            return "long";
+            return ES_TYPE_NAME_LONG;
         case ES_TYPE_HALF_FLOAT:
-            return "half_float";
+            return ES_TYPE_NAME_HALF_FLOAT;
         case ES_TYPE_FLOAT4:
-            return "float";
+            return ES_TYPE_NAME_FLOAT;
         case ES_TYPE_FLOAT8:
-            return "double";
+            return ES_TYPE_NAME_DOUBLE;
         case ES_TYPE_SCALED_FLOAT:
-            return "scaled_float";
+            return ES_TYPE_NAME_SCALED_FLOAT;
         case ES_TYPE_KEYWORD:
-            return "keyword";
+            return ES_TYPE_NAME_KEYWORD;
         case ES_TYPE_TEXT:
-            return "text";
+            return ES_TYPE_NAME_TEXT;
         case ES_TYPE_NESTED:
-            return "nested";
+            return ES_TYPE_NAME_NESTED;
         case ES_TYPE_DATETIME:
-            return "date";
+            return ES_TYPE_NAME_DATE;
         case ES_TYPE_OBJECT:
-            return "object";
+            return ES_TYPE_NAME_OBJECT;
         default:
-            return "unsupported";
+            return "UNSUPPORTED";
```

Comment:
Believe you're missing the VARCHAR type here.

Reply:
Added default and case for VARCHAR type (after catching up with Rupal re: a discussion you had with her)